src/models/room.ts

```diff
@@ -1365,6 +1378,71 @@ export class Room extends EventEmitter {
             // this may be needed to trigger an update.
         }
 
+        if (event.isVisibilityChange()) {
+            // Apply MSC3531-style visibility changes.
+            (() => {
```

Comment:
Why is this a closure? (comments please)

Reply:
Moved to a method, it's more readable that way.